### DISCUSSION TITLE
Optimistic update of many items

### DISCUSSION THREAD

**Discussion by Miloshinjo** [urql-graphql/urql#2541] [Q&A] [answered]
Hello, 

I was wondering if there is something in urql that helps us optimistically update for example, a list of many **checklist items**, if we click them on/off fast, with optimistic updates enabled, it has a weird behavior where the UI changes a lot, which is expected cause as soon as one optimistic update happens and request comes in, another one gets fired etc....

Is there a way to "debounce" this or handle with cache somehow? Anyone had to built something like this before?

Thanks :)

**Comment by kitten** [comment, marked as answer]
I'm not sure if you've checked, but Graphcache keeps structural reference equality as much as possible, hence you don't have to worry about data changing too much and the usual tools to prevent renders like a `useMemo` on some elements or `React.memo` on components works just fine 

I also really wouldn't worry about delaying interactions. Ultimately, if a user clicks and expects a change then you should probably change the UI 😅 I'd only worry if it actually causes performance issues. However, React 18 has more strategies even to make sure that these kinds of rapid interactions don't overwhelm the browser

**Reply by Miloshinjo** [reply]
Thank you for the fast reply :)

**Comment by nandorojo** [comment]
While it isn't an _amazing_ solution, I make the button `disabled={fetching}` in addition to adding the optimistic mutation so that it shows instant feedback but doesn't let the user over-interact with it such that it flickers.

```ts
const [{ fetching }] = useMutation()

<button disabled={fetching}>Like</button>
```
